### ISSUE TITLE
Field loses focus when validation is performed

### ISSUE THREAD

**Issue by mjlime** [Lundalogik/lime-elements#1348]
## Current behavior

![field-loose-focus-after-validation](https://user-images.githubusercontent.com/42939122/132688965-fff1f389-de1a-435e-8231-f9a5aca4e825.gif)

1. Create a schema with a required decimal field.
2. Start with the field empty.
3. Put focus on the field and start typing 1 digit.
4. The field now loses focus when the validation is performed and the error is removed.

## Expected behavior

The focus should still be on the field with the cursor at the same place so I can continue typing seamlessly.

## Environment

- **lime-elements** version: 33.11.0-next.1
- Framework used: 
- Logs:

**Comment by adrianschmidt** [comment]
This is reproducible in the docs for v33.11.0 too. I'd say this is a pretty serious bug.

Thanks for the report!

**Comment by jensgustafsson** [comment]
I could reproduce this with lime-elements v32 as well in the webclient.

**Comment by adrianschmidt** [comment]
> I could reproduce this with lime-elements v32 as well in the webclient.

I've tested a little, and yes, the bug is similar, but not identical in v32.4.4 and the latest version. In v32.4.4, the focus is only lost when the field changes from valid to invalid, whereas in v33.12.0, focus is lost when the validity of the field changes, regardless of whether it's _valid → invalid_ or _invalid → valid_.

I did previously think that the focus problem also applied to non-numeric fields in v33+, but that doesn't seem to be the case. I'm not sure if something's changed, or if I was just confused to begin with 😅

However, the numeric input field in v32.4.4, when configured to format the input, shows the formatted input all the time, instead of showing the actual input when the field is focused. This creates a lag in the input, and other annoying bugs. So, while I think we want to fix the focus bug, because it's really annoying, I don't think it has to stop us from releasing v33+, even if we haven't fixed the bug yet.

@jensgustafsson
